### DISCUSSION TITLE
Relating events over time

### DISCUSSION THREAD

**Discussion by NameOfTheDragon** [datalust/seq-tickets#2123] [Help]
I am gathering log events from a production final test system that records pass/fail results from tests. I have a property `serialNumber` that records the product unique ID, and a property `Disposition` that records the final pass/fail status of the test.

I've been asked if I can show the number of re-tests on a dashboard. So, we would expect to see an event with `Disposition` == `Fail` and then some time later, perhaps days or weeks later, another event with `Disposition` == `Pass`, for the same product serial number.

So I need to gather all the test results for a product for which there is at least one pass result, then count the number of fails and that is the number of re-tests for that particular manufactured item. I can then sum all these up for the day and give a daily running total of re-tests. My SQL skills are not quite up to that, could you suggest a way of doing it, if one exists?

**Comment by nblumhardt** [comment]
Hi @NameOfTheDragon; I don't think this is possible in a straightforward way, unfortunately.

If only a single day were being considered, then:

```sql
select count(Disposition = 'Fail') as fails, count(Disposition = 'Pass') as passes
from stream
group by serialNumber
having fails > 0 and passes > 0
group by time(1d)
```

would give the re-tests performed each day, but only if the fail and subsequent pass were on the same day.

Extracting some raw information and processing it in a separate application might be the only option with this one right now.

Hope this helps,
Nick
